src/util/ActiveShapeUtils.tsx

```diff
@@ -63,3 +78,140 @@ export function Shape<OptionType, ExtraProps, ShapePropsType>({
+    shapeKey = 'trapezoids';
+  } else if (itemDisplayName === 'Pie') {
+    shapeKey = 'sectors';
+  } else if (itemDisplayName === 'Scatter') {
+    shapeKey = 'points';
+  }
+
+  return shapeKey;
+}
+
+/**
+ *
+ * @param {GetActiveShapeIndexForTooltip} arg an object of incoming attributes from Tooltip
+ * @returns {number}
+ *
+ * To handle possible duplicates in the data set,
+ * match both the data value of the active item to a data value on a graph item,
+ * and match the mouse coordinates of the active item to the coordinates of in a particular components shape data.
+ * This assumes equal lengths of shape objects to data items.
+ */
+export function getActiveShapeIndexForTooltip({
+  itemDisplayName,
+  activeTooltipItem,
+  graphicalItem,
+  itemData,
+}: GetActiveShapeIndexForTooltip): number {
+  const shapeKey = getShapeDataKey(itemDisplayName);
+
+  const tooltipPayload: Partial<ShapeData<typeof itemDisplayName>> = isScatter(itemDisplayName, activeTooltipItem)
+    ? activeTooltipItem.payload
+    : activeTooltipItem.tooltipPayload?.[0].payload.payload;
+
+  const activeItemMatches = itemData.filter((datum: any, dataIndex: number) => {
+    const valuesMatch = _.isEqual(tooltipPayload, datum);
+
+    const mouseCoordinateMatches = graphicalItem.props[shapeKey].filter(
+      (shapeData: ShapeData<typeof itemDisplayName>) => {
+        if (isFunnel(itemDisplayName, shapeData) && isFunnel(itemDisplayName, activeTooltipItem)) {
+          const xMatches = shapeData.x === activeTooltipItem?.labelViewBox?.x || shapeData.x === activeTooltipItem.x;
+          const yMatches = shapeData.y === activeTooltipItem?.labelViewBox?.y || shapeData.y === activeTooltipItem.y;
+          return xMatches && yMatches;
+        }
+
+        if (isPie(itemDisplayName, shapeData) && isPie(itemDisplayName, activeTooltipItem)) {
+          const startAngleMatches = shapeData.endAngle === activeTooltipItem.endAngle;
+          const endAngleMatches = shapeData.startAngle === activeTooltipItem.startAngle;
+          return startAngleMatches && endAngleMatches;
+        }
+
+        if (isScatter(itemDisplayName, shapeData) && isScatter(itemDisplayName, activeTooltipItem)) {
+          const xMatches = shapeData.x === activeTooltipItem.x;
+          const yMatches = shapeData.y === activeTooltipItem.y;
+          const zMatches = shapeData.z === activeTooltipItem.z;
+
+          return xMatches && yMatches && zMatches;
+        }
+
+        return false;
+      },
+    );
+
+    // get the last index in case of multiple matches
+    const indexOfMouseCoordinates = graphicalItem.props[shapeKey].indexOf(
+      mouseCoordinateMatches[mouseCoordinateMatches.length - 1],
+    );
+
+    const coordinatesMatch = dataIndex === indexOfMouseCoordinates;
+
+    return valuesMatch && coordinatesMatch;
+  });
+
+  // get the last index in case of multiple matches
+  const activeIndex = itemData.indexOf(activeItemMatches[activeItemMatches.length - 1]);
```

Comment:
what if length is 0?

Reply:
We'd get `activeIndex` as `-1`, i.e. no match. Which i think is what we'd want for a fallback, correct?

Reply:
when length is zero this happens no?
`activeItemMatches[0 - 1]` === `activeItemMatches[-1]` === out of bounds exception

Reply:
Oh wow javascript continues to surprise me, I guess that just returns undefined huh 🙃

Reply:
All good, ty